inc/event_listeners.h

```diff
@@ -0,0 +1,48 @@
+#ifndef EVENT_LISTENERS_H
+#define EVENT_LISTENERS_H
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <tuple>
+
+enum Event {
+  BEGIN_PHASE,
+  DO_PREDICT_BRANCH
+};
+
+#include "listeners/cpi_printout.h"
+#include "listeners/test_listener.h"
+
+inline std::vector<bool> listener_activation_map;
+inline auto listeners = std::make_tuple(new CPIPrintout(), new TestListener());

```

Comment:
```suggestion
inline auto listeners = std::make_tuple(new CPIPrintout(), new TestListener());
inline std::bitset<std::tuple_size_v<decltype(listeners)>> listener_activation_map;
```

Might as well do something like this since the size is statically known.